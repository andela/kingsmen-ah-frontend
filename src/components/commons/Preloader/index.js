import React from 'react';
import Loader from 'react-loader-spinner';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import './index.scss';


function Preloader(props) {
  const { type, styles, height, width, color } = props;
  const className = classNames({
    'main-preloader center justify-center items-center': type === 'page',
    '': type === 'button'
  });

  return (
    <div className={className}>
      <Loader type={styles} height={height} width={width} color={color} className="text-center mx-auto center" />
    </div>
  );
}

Preloader.propTypes = {
  type: PropTypes.string,
  styles: PropTypes.string,
  height: PropTypes.number,
  width: PropTypes.number,
  color: PropTypes.string,
};

Preloader.defaultProps = {
  type: null,
  styles: null,
  height: null,
  width: null,
  color: null
};

export default Preloader;
