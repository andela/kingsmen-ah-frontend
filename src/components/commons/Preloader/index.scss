$loader-bg: rgb(233, 228, 228);

.main-preloader {
  div {
    left: 50%;
    position: absolute;
    top: 50%;
    transform: translate(-50%, -50%);
  }
  background: $loader-bg;
  bottom: 0;
  height: 100vh;
  left: 0;
  opacity: 0.5;
  position: absolute;
  right: 0;
  top: 0;
}

svg {
  margin: 0 auto;
}

.main-preloader div {
  left: 50%;
  position: absolute;
  top: 50%;
  transform: translate(-50%, -50%);
}
