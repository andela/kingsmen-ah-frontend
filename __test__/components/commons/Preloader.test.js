import React from 'react';
import Preloader from '@components/commons/Preloader';


const props = {
    type: 'button',
    height: 40,
    width: 40,
    style: 'TailSpin',
    color: 'green'
};

describe('<Preloader />', () => {
    it('should render a preloader with type ~ button', () => {
        const shallowWrapper = shallow(<Preloader {...props} />);

        expect(toJson(shallowWrapper)).toMatchSnapshot();
        expect(shallowWrapper.find('Preloader')).toBeTruthy();
    });

    it('should render a preloader with type ~ main-page', () => {
        const shallowWrapper = shallow(<Preloader {...props} type="main-preloder" />);

        expect(toJson(shallowWrapper)).toMatchSnapshot();
        expect(shallowWrapper.find('Preloader')).toBeTruthy();
        expect(shallowWrapper.find('main-preloder')).toBeTruthy();
    });
});
