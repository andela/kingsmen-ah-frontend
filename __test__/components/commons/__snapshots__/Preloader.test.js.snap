// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Preloader /> should render a preloader with type ~ button 1`] = `
<div
  className=""
>
  <Loader
    className="text-center mx-auto center"
    color="green"
    height={40}
    type={null}
    width={40}
  />
</div>
`;

exports[`<Preloader /> should render a preloader with type ~ main-page 1`] = `
<div
  className=""
>
  <Loader
    className="text-center mx-auto center"
    color="green"
    height={40}
    type={null}
    width={40}
  />
</div>
`;
